webviz_subsurface/private_containers/_surface_selector.py

```diff
@@ -0,0 +1,586 @@
+
+Creates a widget to select surfaces from a yaml configuration file or dictionary, and
+a dataframe of ensemble/realizations, optionally with sensitivity cases.
+The current selections are stored in a dcc.Store object that can
+be accessed by the storage_id property of the class instance.
+
+* `config`: A dictionary / yaml configuration file of surfaces on the format below
+* `ensembles`: A pandas dataframe with ensemble, real(index), runpath, sensname and senscase
+
+Format of configuration:
+some_property:
+    names:
+        - surfacename
+        - surfacename
+    dates:
+        - somedate
+        - somedate
+another_property:
+    names:
+        - surfacename
+        - surfacename
+    dates:
+        - somedate
+        - somedate
+"""
+
+    def __init__(self, app, config, ensembles):
+        self._configuration = self.read_config(config)
+        self._ensembles = ensembles
+        self._storage_id = f"{str(uuid4())}-surface-selector"
+        self.set_ids()
+        self.set_callbacks(app)
+
+    @staticmethod
+    def read_config(config):
+        """Reads config file either from a yaml provided file or from a dict"""
+        if isinstance(config, str):
+            return yaml.safe_load(open(config, "r"))
+        elif isinstance(config, dict):
+            return config
+        else:
+            raise TypeError("Config must be a dictionary of a yaml file")
+
+    @property
+    def storage_id(self):
+        """The id of the dcc.Store component that holds the selection"""
+        return self._storage_id
+
+    def set_ids(self):
+        uuid = str(uuid4())
+        self.attr_id = f"{uuid}-attr"
+        self.attr_id_btn_prev = f"{uuid}-attr-btn-prev"
+        self.attr_id_btn_next = f"{uuid}-attr-btn-next"
+        self.name_id = f"{uuid}-name"
+        self.name_id_btn_prev = f"{uuid}-name-btn-prev"
+        self.name_id_btn_next = f"{uuid}-name-btn-next"
+        self.date_id = f"{uuid}-date"
+        self.date_id_btn_prev = f"{uuid}-date-btn-prev"
+        self.date_id_btn_next = f"{uuid}-date-btn-next"
+        self.ensemble_id = f"{uuid}-ens"
+        self.ensemble_id_btn_prev = f"{uuid}-ensemble-btn-prev"
+        self.ensemble_id_btn_next = f"{uuid}-ensemble-btn-next"
+        self.realization_id = f"{uuid}-real"
+        self.realization_id_btn_prev = f"{uuid}-realization-btn-prev"
+        self.realization_id_btn_next = f"{uuid}-realization-btn-next"
+        self.attr_wrapper_id = f"{uuid}-attr-wrapper"
+        self.name_wrapper_id = f"{uuid}-name-wrapper"
+        self.date_wrapper_id = f"{uuid}-date-wrapper"
+        self.ens_wrapper_id = f"{uuid}-ens-wrapper"
+        self.real_wrapper_id = f"{uuid}-real-wrapper"
+        self.aggreal_id = f"{uuid}-aggreal"
+        self.sens_name_id = f"{uuid}-sens-name-id"
+        self.sens_case_id = f"{uuid}-sens-case-id"
+        self.sens_name_wrapper_id = f"{uuid}-sens-name-wrapper-id"
+        self.sens_case_wrapper_id = f"{uuid}-sens-case-wrapper-id"
```

Comment:
Some of the IDs are not used in callbacks, then they can perhaps be removed altogether? Both here + in the component they are currently used in?

Reply:
I could only find one that is not used. Removed that one :)